_posts/2022-11-20-point-in-time.md

```diff
@@ -0,0 +1,232 @@
+
+When you use a query with a PIT ID, it searches the segments that are frozen in time. Because a PIT is query-agnostic, you can use any query to search the data in a PIT. PIT allows for consistent pagination because even though the index continues to ingest and delete documents, the PIT does not reflect those changes and the dataset remains constant. Alternatively, if you use a normal query without a PIT ID, it searches live segments. 
+
+### What's the catch?
+
+So far we've seen that PIT search is superior to other pagination methods. But what are the drawbacks? First, for a PIT, OpenSearch has to keep the segments even though they might have been merged and are not needed for the live dataset. This leads to an increased heap usage. Second, there is currently no built-in resiliency in PIT, so if your node goes down, all PIT segments are lost.
+
+## How to use PIT
+
+The example in this section uses the `shakespeare` index. 
+
+### Set up a sample index
+
+Assuming you are not running the security plugin, you can set up the `shakespeare` index as follows:
+
+1. Download the mapping file: 
+    ```bash
+    wget http://media.sundog-soft.com/es7/shakes-mapping.json
+    ```
+
+1. Index the mapping file:
+    ```bash
+    curl -H "Content-Type: application/json" -XPUT http://localhost:9200/shakespeare (https://localhost:9200/shakespeare) --data-binary "@shakes-mapping.json"
+    ```
+
+1. Download the `shakespeare` dataset:
+    ```bash
+    wget http://media.sundog-soft.com/es7/shakespeare_7.0.json
+    ```
+
+1. Index the data into OpenSearch:
+    ```bash
+    curl -H "Content-Type: application/json" -XPUT http://localhost:9200/shakespeare/_bulk (https://localhost:9200/shakespeare/_bulk) --data-binary "@shakespeare_7.0.json"
+    ```
+
+### Use the PIT functionality
+
+Follow these steps to use PIT. 
+
+**Step 1: Create a PIT**
+
+The following request creates a PIT that will be kept for 1 hour:
+
+```json
+POST shakespeare/_search/point_in_time?keep_alive=1h
+```
+
+The response contains a PIT ID:
+
+```json
+{
+  "pit_id" : "o8L8QAELc2hha2VzcGVhcmUWck5NRDE3SFNUZ3lRZmJEMW1COWRidwAWd21kTFZRWlZSQ2k2YmRDeWh4U2w3ZwAAAAAAAAAACBZ0aFdURzJZbVJIYVlxczBsbkZ4emVnARZyTk1EMTdIU1RneVFmYkQxbUI5ZGJ3AAA=",
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "creation_time" : 1669082665118
+}
+```
+
+**Step 2: Use the PIT ID for your search**
+
+The following request searches for all documents in the play "Hamlet," sorted by speech number and then id, and returns the first 20 results:
+
+```json
+GET /_search
+{
+  "size": 20,
+  "query": {
+    "match" : {
+      "play_name" : "Hamlet"
+    }
+  },
+  "pit": {
+    "id":  "o8L8QAELc2hha2VzcGVhcmUWck5NRDE3SFNUZ3lRZmJEMW1COWRidwAWd21kTFZRWlZSQ2k2YmRDeWh4U2w3ZwAAAAAAAAAACBZ0aFdURzJZbVJIYVlxczBsbkZ4emVnARZyTk1EMTdIU1RneVFmYkQxbUI5ZGJ3AAA=", 
+    "keep_alive": "100m"
+  },
+  "sort": [ 
+    { "speech_number": "asc" },
+    { "_id": "asc" }
+  ]
+}
+```
```

Comment:
Is it mandatory, I guess not, should we explain what happens if we haven't specified a sort order?

Reply:
I think this is just a concept overview; they can go to the documentation for details, defaults, etc.